### DISCUSSION TITLE
Error:The server could not connect to validation target

### DISCUSSION THREAD

**Discussion by sferris** [smallstep/certificates#784] [Q&A] [answered]
Hi all -- I'm having a heck of a time trying to get the acme interface working.. I'm running step-ca in a podman container, with --net=host, and a raspberry pi, and not rootless. (though if I can get it running, I would like to switch eventually)

Here's the command that I'm running from my laptop: (actual domain name replaced)

```
$ /opt/step_0.18.0/bin/step ca certificate lucy.foo lucy.crt lucy.key --provisioner acme --http-listen :8080
```

And it takes a bit to return the error:

```
Unable to validate challenge: &{ID: AccountID: AuthorizationID: Value: Type:http-01 Status:pending Token:j9OmN5gb0*******************zydK8N ValidatedAt: URL:https://ca.foo:9000/acme/acme/challenge/mKWu5xFwwIQuZnqPTyleS6O1In9EqblT/VOFN2qnzV5sN6RhJu401ax8F80BlhPQi Error:The server could not connect to validation target}
```

From my laptop I can reach the ca without issue (system certs installed):

```
$ curl https://ca.foo:9000/health
{"status":"ok"}
```

As well, when the command is running I can both resolve the ip of my laptop, as well, connect to the listener port 8080.. from within the container:

```
# nslookup lucy.foo
Server:         192.168.0.5
Address:        192.168.0.5:53

Name:   lucy.foo
Address: 192.168.0.196
```

```
$ telnet lucy.foo 8080
Connected to lucy.foo
^]

Console escape. Commands are:

 l      go to line mode
 c      go to character mode
 z      suspend telnet
 e      exit telnet
e
$        
```

Any help on what I'm missing?! I'm at a loss.. Appreciate any help! :D (Happy Holidays!)

**Comment by tashian** [comment, marked as answer]
Hi @sferris, the ACME HTTP challenge requires the client to be able to serve the challenge response on port 80, because port 80 is a privileged port. This way, only privileged users can get a certificate signed for the host. The purpose of the `http-listen` flag is to run the local ACME validation server on a non-standard port. But, it's only to be used behind a reverse proxy that would, in turn, make the challenge response available on port 80.

**Reply by sferris** [reply]
Well damn.. I knew that was a stipulation for letsencrypt, and yet, I still completely ignored it when I saw the option.. That worked:

```
$ sudo caddy reverse-proxy --from http://lucy.foo --to localhost:8080 
```

I appreciate the help.. tyvm!
